zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationConfig.java

```diff
@@ -330,6 +353,13 @@ public String getZnodeGroupAclServerDedicatedDomain() {
     return znodeGroupAclServerDedicatedDomain;
   }
 
+  public boolean shouldEnforceDedicatedDomain() {
+    if (znodeGroupAclServerShouldEnforceDedicatedDomain == null) {
+      setZnodeGroupAclServerShouldEnforceDedicatedDomain(System.getProperty(ENFORCE_DEDICATED_DOMAIN, "true"));
```

Comment:
nit: you can directly use your instance variable?

Reply:
If you want to follow rest of the variables pattern then we should lazy bind it in getter method and have only one setter method. I suggest we should do it so that we can keep it consistent and it will also keep setter method names consistent.

Reply:
`shouldEnforceDedicatedDomain()` is the getter method and is already following the convention you mentioned. Am I missing something?